azure-pipelines-official.yml

```diff
@@ -37,9 +37,16 @@ stages:
     - powershell: Write-Host "##vso[task.setvariable variable=SourceBranchName]$('$(Build.SourceBranch)'.Substring('refs/heads/'.Length))"
       displayName: Setting SourceBranchName variable
 
+    - powershell: Write-Host "##vso[task.setvariable variable=NonVSDepsBranch]$($Build.SourceBranch -like '*-vs-deps')"
+      displayName: Setting VSDepsBranch variable
+
     - powershell: Write-Host "##vso[task.setvariable variable=VisualStudio.DropName]Products/$(System.TeamProject)/$(Build.Repository.Name)/$(SourceBranchName)/$(Build.BuildNumber)"
       displayName: Setting VisualStudio.DropName variable
 
+    - powershell: git branch -r | select-string "origin/$(Build.SourceBranch)-vs-deps" | select -First 1 | %{ Write-Host "git merge $_" }
+      displayName: Merging VS Deps 
+      condition: and(succeeded(), not(variables['VSDepsBranch']))
```

Comment:
This will essentially gate the reliability of our official builds on the ability to blindly merge the vs deps branch correct?

Reply:
Maybe we should have another variable that lets us opt into this behavior, e.g. `MergeVsDeps`. Since somewhere down the line we may even get into a situation where a *-vs-deps branch exists, but we don't actually want to merge it when running the signed build.

Reply:
I like the idea of another variable. Should that be defined in the yaml or azdo directly?

Reply:
I think it'd be in azdo, same as `SignType`, `IbcDropId`, etc.

Reply:
Note that this is already conditional on the `VSDepsBranch` variable, but adding it as an explicit variable to azdo makes sense.